lib/temporal/workflow/context.rb

```diff
@@ -17,6 +17,24 @@ class Workflow
     class Context
       attr_reader :metadata, :config
 
+      module Private
+        # Shared with LocalWorkflowContext so we can do the same validations in test and production.
```

Comment:
Since it is shared we need to move it out into its own file, otherwise you're depending `LocalContext` on the full `Context`